### ISSUE TITLE
How to use/run AnnotationSpec style tests?

### ISSUE THREAD

**Issue by crazyk2** [kotest/kotest#1298]
Your [example](https://github.com/kotest/kotest/blob/master/kotest-tests/kotest-tests-core/src/jvmTest/kotlin/com/sksamuel/kotest/specs/annotation/AnnotationSpecExample.kt) doesn't work.
If I run [FunSpecExample](https://github.com/kotest/kotest/blob/master/kotest-tests/kotest-tests-core/src/jvmTest/kotlin/com/sksamuel/kotest/specs/funspec/FunSpecExample.kt) tests all works well.
How to run\configure AnnotationSpec tests correctly?

**Comment by sksamuel** [comment]
What version ?

**Comment by crazyk2** [comment]
> What version ?

Latest v4.0.0-BETA3

**Comment by crazyk2** [comment]
@sksamuel I found a simple way to fix this issue. I could send PR but if you rewrite base code (as in your [commit](https://github.com/kotest/kotest/commit/715eacbc0f4572613c86928a34bb93de15b51218)) I could just wait)))

**Comment by sksamuel** [comment]
It's fixed in master now.

**Comment by crazyk2** [comment]
I read your commit. Great changes to better readability and structure. But why you made difference between Annotation and DSL styles in code. 
The problem was that tests weren't add to rootTestCases because tests were written outside `init{}` block if compare with other styles when instance of class is initializing by call [createInstance](https://github.com/kotest/kotest/blob/715eacbc0f4572613c86928a34bb93de15b51218/kotest-core/src/jvmMain/kotlin/io/kotest/core/engine/SpecExecutor.kt#L87) -> [instantiateSpec](https://github.com/kotest/kotest/blob/715eacbc0f4572613c86928a34bb93de15b51218/kotest-core/src/jvmMain/kotlin/io/kotest/core/engine/instantiateSpec.kt#L13). Because all test classes are child of AnnotationSpec I fix issue in the way add `init{}` block to this class with code
```kotlin
  init{
      this::class.findTestFunctions().map {
         if (it.isIgnoredTest()) {
            it.toIgnoredTestCase()
         } else {
            it.toEnabledTestCase()
         }
      }
   }
```
without your latest changes of course and all work well. I think there is no reason to create `DslDrivenSpec` and  `rootTestCases, addRootTestCase` work for AnnotationSpec well

**Comment by sksamuel** [comment]
I think my way is clearer that with dsl specs the tests are added in place
and with annotation specs they are created when asked.

**Comment by crazyk2** [comment]
Did you mean `TestCase`? Because test classes are initialized any way

**Comment by crazyk2** [comment]
I am a new in kotest and it would be great to have common workflow of tests\big schema to understand how it works. Maybe do you have something like that?

**Comment by sksamuel** [comment]
Are you asking which styles to use when laying out the tests ?

**Comment by crazyk2** [comment]
> Are you asking which styles to use when laying out the tests ?

No I mean engine under the hood. How tests group, how runs how separated on threads and etc. Because now its a big work to read all framework to understand details. in past time I did that work with testNG and know pros and cons of this legacy framework (I called "legacy" because now exists kotest;) ) and I wanna help or adapt framework for my purpose
